packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,401 @@
+        uint256 minApprovals;
+        uint64 snapshotBlock;
+    }
+
+    /// @notice A container for the proposal tally.
+    /// @param approvals The number of approvals casted.
+    /// @param addresslistLength The length of the .
+    struct Tally {
+        uint256 approvals;
+        uint256 addresslistLength;
+    }
+
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
+    bytes4 internal constant MULTISIG_INTERFACE_ID =
+        this.addAddresses.selector ^
+            this.removeAddresses.selector ^
+            this.isListed.selector ^
+            this.addresslistLength.selector ^
+            this.initialize.selector;
+
+    /// @notice The ID of the permission required to call the `addAddresses` and `removeAddresses` functions.
+    bytes32 public constant UPDATE_MULTISIG_SETTINGS_PERMISSION_ID =
+        keccak256("UPDATE_MULTISIG_SETTINGS_PERMISSION");
+
+    /// @notice The minimum approval parameter.
+    uint256 private minApprovals_;
+
+    /// @notice A mapping between proposal IDs and proposal information.
+    mapping(uint256 => Proposal) internal proposals;
+
+    /// @notice Thrown if a approver is not allowed to cast a approve. This can be because the proposal
+    /// - is not open,
+    /// - was executed, or
+    /// - the approver is not on the addresslist
+    /// @param proposalId The ID of the proposal.
+    /// @param sender The address of the sender.
+    error ApprovalCastForbidden(uint256 proposalId, address sender);
+
+    /// @notice Thrown if the proposal execution is forbidden.
+    /// @param proposalId The ID of the proposal.
+    error ProposalExecutionForbidden(uint256 proposalId);
+
+    /// @notice Thrown if the minimal approvals value is out of bounds (less than 1 or greater than the number of members in the address list).
+    /// @param limit The maximal value.
+    /// @param actual The actual value.
+    error MinApprovalsOutOfBounds(uint256 limit, uint256 actual);
+
+    /// @notice Emitted when new members are added to the address list.
+    /// @param members The array of member addresses to be added.
+    event AddressesAdded(address[] members);
+
+    /// @notice Emitted when members are removed from the address list.
+    /// @param members The array of member addresses to be removed.
+    event AddressesRemoved(address[] members);
+
+    /// @notice Emitted when the multisig settings are updated.
+    /// @param minApprovals The minimum approvals value.
+    event MinApprovalUpdated(uint256 minApprovals);
+
+    /// @notice Emitted when an proposal is approve by an approver.
+    /// @param proposalId The ID of the proposal.
+    /// @param approver The approver casting the approve.
+    event Approved(uint256 indexed proposalId, address indexed approver);
+
+    /// @notice Thrown when a sender is not allowed to create a approve.
+    /// @param sender The sender address.
+    error ProposalCreationForbidden(address sender);
```

Comment:
The order of this error is a bit off, why it comes after the events, is it intentional

Reply:
Fixed in d1af8c2befc10c9aeff7605cf0ffefef34e157d5